apis/v1alpha1/clusterprofile_types.go

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ClusterProfileSpec defines the desired state of ClusterProfile.
+type ClusterProfileSpec struct {
+	// DisplayName defines a human-readable name of the ClusterProfile
+	// +optional
+	DisplayName string `json:"displayName"`
+
+	// ClusterManager defines which cluster manager owns this ClusterProfile resource
+	// +required
+	ClusterManager ClusterManager `json:"clusterManager"`
+}
+
+// ClusterManager defines which cluster manager owns this ClusterProfile resource.
+// This field is immutable.
+// It's recommended that each cluster manager instance should set a different values to this field.
+// In addition, it's recommended that a predefined label with key "x-k8s.io/cluster-manager"
+// should be added by the cluster manager upon creation. See constant LabelClusterManager.
+// The value of the label should be the same as the name of the cluster manager.
+// The purpose of this label is to make filter clusters from different cluster managers easier.
+//
+// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="ClusterManager is immutable"
+type ClusterManager struct {
+	// Name defines the name of the cluster manager
+	// +required
+	Name string `json:"name"`
+}
+
+// ClusterProfileStatus defines the observed state of ClusterProfile.
+type ClusterProfileStatus struct {
+	// Conditions contains the different condition statuses for this cluster.
+	Conditions []metav1.Condition `json:"conditions"`
+
+	// Version defines the version information of the cluster.
+	Version ClusterVersion `json:"version,omitempty"`
+
+	// Properties defines properties of collected from the cluster.
+	// For example, a unique cluster identifier (id.k8s.io).
+	// The set of properties is not uniform across a fleet, some properties can be
+	// vendor or version specific and may not be included from all clusters.
+	// +optional
+	Properties []Property `json:"properties,omitempty"`
+}
+
+// ClusterVersion represents version information about the cluster.
+type ClusterVersion struct {
+	// Kubernetes is the kubernetes version of the cluster.
+	// +optional
+	Kubernetes string `json:"kubernetes,omitempty"`
+}
+
+// Property defines a property collected from a cluster.
+type Property struct {
```

Comment:
This was heavily debated during a few SIG MC meetings. If we include real time data such as CPU, GPU, memory etc. it might not scale well when the cluster manager watches many clusters.
For now, I've adjusted the comment to mention that it may include these data because I think it's still up the user of this API. WDYT?

Reply:
Good point Mike. I've reworked the description to mention that it could be a ClusterProperty (or something else). PTAL thanks.

Reply:
The current thinking is to keep the property field open while the community comes to an agreement on what are the "core" properties every provider should support.  There are also "nice to have", "you are on your own" properties that we can list in the KEP/CRD.

Reply:
Revised description looks good enough to get started, okay to resolve this thread I think!